### DISCUSSION TITLE
SX126x.cpp SetTx() latency

### DISCUSSION THREAD

**Discussion by mgergos** [jgromes/RadioLib#1008] [General]
Hi Jan,

Would it be possible to add a build option to utilize RADIOLIB_SX126X_STANDBY_XOSC?
The standby() function currently defaults to _RC adding around 5 ms to each SetTx() call.

It could replace any hardcoded standby() calls with standby(RADIOLIB_SX126X_STANDBY_XOSC).
Additionally,  the fallback mode would need to be RADIOLIB_SX126X_RX_TX_FALLBACK_MODE_STDBY_XOSC within config().

Maybe there's a more elegant solution?
Let me know your thoughts.

M.

**Comment by jgromes** [comment]
I think this can be done, either as a build option as you suggested, or a runtime flag (similar to `SX126x::XTAL`). I'm actually more in favor of the second approach, since it lets the user create two different SX126x instances with two different standby configurations. I can't immediately think of a use case for that, but it seems there could be some.

I propose to add `bool SX126x::standbyXOSC` member which defaults to false. If set to true, it will use the XOSC standby mode. Will that work for you?

**Reply by mgergos** [reply]
Sounds good!
No rush.

**Reply by jgromes** [reply]
Added in b288485d6c43feddb1c56a66d75ae08e7e8defab, will be released as part of the 6.4.3 update in the coming days hopefully.

**Reply by mgergos** [reply]
Just one more edit please Jan (SX126x.cpp line 2090);
`data[0] = RADIOLIB_SX126X_RX_TX_FALLBACK_MODE_STDBY_RC;`
Should be;
`data[0] = this->standbyXOSC ? RADIOLIB_SX126X_RX_TX_FALLBACK_MODE_STDBY_XOSC : RADIOLIB_SX126X_RX_TX_FALLBACK_MODE_STDBY_RC;`

**Reply by jgromes** [reply]
Ah, sorry - forgot about the fallback mode. Fixed in 9fd7db4d13ada0f7681900ff8ab3512d93440fc5

**Reply by mgergos** [reply]
Awesome Jan, that's working perfectly now. Thanks.
For the benefit of others, `radio.standbyXOSC = true;` should be set prior to `radio.begin();` 
(Should your radio application require speed over power efficiency).
